### DISCUSSION TITLE
EKHHE260CV37 unit support

### DISCUSSION THREAD

**Discussion by D1ang** [raomin/ESPAltherma#162] [Ideas]
I have a Daikin domestic water heat pump (EKHHE260CV37) only for hot showers and was wondering if it is supported.
I've been looking for a X10 port but haven't found any (yet) But I do have some pictures of the board.

There is a diagram in the manual, but I'm not sure where to start.
So I was wondering if anyone here has the same unit and success conecting it?

![IMG_20221007_151140](https://user-images.githubusercontent.com/52288307/194635968-e3e380be-2e6d-4283-bb5d-f661e8e2059b.jpg)
![IMG_20221007_150924](https://user-images.githubusercontent.com/52288307/194635975-0be150db-6dce-42d0-9da3-4789fc114644.jpg)
![IMG_20221007_150042](https://user-images.githubusercontent.com/52288307/194635978-5c683847-1c9d-424a-8975-2231a9fb7f19.jpg)

**Comment by raomin** [comment]
It's unlikely that it behaves like an Altherma heat pump but you can try on this red port. 

If not, connect it to a serial <-> USB and have a look with a serial terminal to see what's coming out.

**Reply by D1ang** [reply]
Thanx I'll try that asap. See what will happen, wish me luck :p

**Comment by assembly12** [comment]
Hi @D1ang ,
I'm in the process of buying a heatpump for **domestic** hot water and was baffled by the lack of smarthome/wifi support.
The Daikin model you mention is the only model which at least has a connector for a Wifi-module, but no further information as to what kind of module and the possibilities is available. This Daikin model is however now on the top of my favoured models.

Were you able to create a serial connection? If so, what does is look like?
I have some experience with reverse engineering modbus/serial connections and as long as there is communation on this port, i'm sure it will be possible to interpret at least some of it.

**Reply by D1ang** [reply]
Hi @assembly12, 
No I hadn't tried it yet. Some other stuff came in between.
I'm not that experienced (yet) with reverse engineering modbus/serial connections.
But I'm still planning to try this out, not sure when.

**Reply by assembly12** [reply]
Hi @D1ang,
I've ordered the unit and it arrived yesterday. I'm not sure when I get around to installing it (probably in march/april), but will definitely have a go at making it smart. I will post my findings here or perhaps start a new thread on the HA forums.
Before that though, I will defenitely try if ESPALTHERMA is compatible. It's unlikely but not totally out of the question and worth a shot.

**Comment by KalaNagHTD** [comment]
![60ABE744-45D2-45FC-BAB4-393F4A8557F8](https://user-images.githubusercontent.com/77481230/212045584-76a5fd6f-3e4b-4d2f-b456-2cee5ad425c1.jpeg)

**Comment by assembly12** [comment]
Thanks. Yeah I know that diagramm from the installation manual. It's basicly the only info you can find on the subject. Daikin doesn't even advertise the possibility that wifi might be added on CN22. I've written german and europe customer support a couple weeks ago, but didn't receive any answer.

**Comment by blablaaddi** [comment]
would love to get mine online as as well.

**Comment by hwinkel** [comment]
I have written them several times as well, about the WiFi Module. Never got any answer from DAIKIN. Where it is unlikely you get any support from such vendor.

BTW: I have writtten to Dimplex, Remeha, Wolf, Waterkotte, AIT (Nibe, Alpha Innnotec), Austria Email... ALL of them responded within days. and I receoed the infos about integration. Some have direct Modbus Support and even send all the Technical Material

**Comment by hwinkel** [comment]
but at least it seems to be a Serial Rx/Tx connection on CN22 (and CN21)

**Comment by nwd61** [comment]
I recently purchased the Daikin Altherma M HW 200 EKHHE200CV37 hot water heat pump. 
I believe it has the same electronics as the EKHHE260CV37. 
I am also looking for a way to read and possibly log the sensor data from it, and perhaps make some changes. 
Have there been any developments or additional information from you on how to do this?

**Reply by assembly12** [reply]
I still intend to get this working. However up till now, my hot water heat pump isn't even installed yet.
I hope I'll get it done this year still.

**Comment by toastbroat13** [comment]
Hi all.

My Daikin Altherma EKHHE260CV37 is already running and I was also "surprised" that there is no official way to connect the "SG ready" and "WIFI" Daikin to the network... :-(

It would be very nice to find a solution to connect and control the heat pumb - in summer season to use solar power and in winter season to use cheap power via Tibber...

So I would love to hear any updates abot ESPAltherma and the Daikin EKHHE units.

Many thanks in advance and kind regards,
Matthias

**Comment by hwinkel** [comment]
Seems we are all in the same boat. The situation with the warm water heatpumps is the same across vendors. There are not well supported in terms of interfaces for control

**Comment by toastbroat13** [comment]
Doas anybody know, which DAIKIN WIFI or LAN Module would work? I tried to search and found several "informations" about possible module which might work but at DAIKIN I didn't found an official information. 

The board includes the WIFI Card Connection (not included) at CN 22 - so it seems there should be some official adon to connect the wwhp to the local network...

**Reply by nwd61** [reply]
As you mentioned, Installation Manual says: "CN22  : WI-FI card connection (not supplied)"
Seems that there is no suitable DAIKIN-CARD. Maybe, that someone has luck with a blind shot with ESPAltherma?

**Comment by donwolle** [comment]
did anybody test the combination or do i have to be the first one?

**Reply by hwinkel** [reply]
As being in the threat, just want to inform you that I have decided to go with a heat pump with build in Modbus support.

**Comment by nwd61** [comment]
@donwolle: It would be great if you could do this test and analysis, e.g. with an Arduino, and share your values here. We may then be able to help you analyze the scanned values.

**Comment by Hansfreddy** [comment]
Did anyone ever get anything out of these 2 connections (or even the connection for the display)? Would be willing to hook a USB-Serial converter up to it but only if no one has yet confirmed defeat :)

**Reply by jcappaert** [reply]
I have an EKHHE260CV37 and I have attempted briefly on both connectors (beware, pinouts are different) with a USB-Serial converter and cycled through some settings (baud rate, parity, stop bits), but have gotten nothing from it. I also tried to send some characters to see if that prompted anything and have also had it connected during power cycle but still nothing. Most of my attempts were on CN22, I just did a basic check on CN21. I wouldn't call it defeat yet, but at least it's not obvious :)

**Reply by Hansfreddy** [reply]
I see. 
Maybe the best bet would be to check CN23 (display) because we know there is data flowing. I'll go search my USB-Serial converter and have a go.

Last resort is to read the display with ESP32-CAM and manipulate the buttons with an ESP32.

**Comment by Hansfreddy** [comment]
Yesterday I opened my water heater and saw that there was a cable already connected to CN21 (in the manual described as F. C. T - possibly field connected terminal). Enthusiastically, I connected my scope to it and saw - nothing! I'ts possible that I did something wrong, or that the port only works while programming or that there is a hidden option - we don't know. However, I will not further go down this route.

Instead, I will try to analyze the SPI interface (CN17) - maybe some information can be retrieved from there, although the port seems undocumented.

If that doesn't work out, we're back to interfacing the waterheater externally (actuators on the touch buttons and esp32 to read the display). I think that should be feasible, but of course it would be nicer to get the data "directly" from the device.

I'll keep you posted

**Reply by jcappaert** [reply]
FYI on my unit the CN21 cable is just connected to a cable stub that runs behind the PCB area but it's not actually plugged into anything. Presumably left there indeed for programming or maintenance.

**Reply by Hansfreddy** [reply]
It was the same for my unit. But I took it as a good sign that the stub was there - which meant to me that it must be good for something when they bothered leaving it there ;)

**Comment by Hansfreddy** [comment]
@jcappaert @roboticsalign
Just wanted to give a quick update: I had to order some parts to analyze the PCB and Protocol further.
I still think the SPI interface is our best bet but I'll check back with you!

**Reply by jcappaert** [reply]
Happy to try and measure or reproduce anything if it's helpful, I've got some logic analyzers and other stuff here ...

**Comment by Hansfreddy** [comment]
Wanted to check in back here. I did not have time yet for the logic analysis, but at least today I built the level shifter (Daikin PCB is 12V, logic analyzer is 5V). All the while, I thought about Plan B and came to the conclusion, that "digital" control of the water heater isn't really needed:

- As far as I know, @roboticsalign has put a relay in line with the resisitve heating module (CN16) so we can control, if the heater is used or not.
- With Digital Input 1 we can send the heater into "overdrive", heating till the configured threshold
- With Digital Input 2 we can deactivate the water heater completely

We could then configure the water heater to "auto mode" and 62 degrees C target temperature. Using the parameters mentioned we could fully control it. There really are only 2 things missing: Temperature Info (maybe we get it from SPI, otherwise ESP32-Cam or 2 additional temperature sensors) and adding the in-line relay.

Regarding the heating element: In auto mode, the water heater sometimes activates the heating element. It would be nice to not only control this but also being able to detect, when the water heater would "like" to activate the module.

Maybe you can help me out there: I would like to disconnect the heating element from CN16 and replace with a 1 MOhm resistor. As soon as the Daikin relay connects, a current will flow (very little, 0,00023A about 0,05 watts). This I could then detect in an ESP32 and trigger another relay where the heating element will be connected instead. 
What would be your approach to solving this?

**Reply by roboticsalign** [reply]
Two thoughts:

- From my understanding, you cannot use DIG1 and DIG2 together. That's why I operate it in boost mode. 
- you could use the switch input of a shelly 1 to detect behavior of the Daikin. (and you can flash a Shelly with Esphome, if needed)

**Reply by lorbetzki** [reply]
Hello,

thank your for your information, i create another repository today because i can receive data over CN23. This is Modbus RTU with 9600 baud. If you like, we can continue discussing here and find out how the data is made up.

https://github.com/lorbetzki/Daikin-EKHHE

Best regards

**Reply by Hansfreddy** [reply]
> Hello,
> 
> thank your for your information, i create another repository today because i can receive data over CN23. This is Modbus RTU with 9600 baud. If you like, we can continue discussing here and find out how the data is made up.
> 
> https://github.com/lorbetzki/Daikin-EKHHE
> 
> Best regards

This is very interesting. My first idea was also to attach to the display but I was afraid to brick the device. Now that I know that we can connect in parallel this seems like the way to go. In theory, we should be able to do everything the display can do, just need to figure out the data...

**Reply by Hansfreddy** [reply]
Okay so I tried to capture some traffic with my logic analyzer. Unfortunately, I got errors in sigrok, so I guess I did something wrong - it's really not my field of expertise.

I then tried to connect to the SPI Port but got no data at all. I guess it needs to be activated, maybe by use of CN21 but that's guesswork and after all - I don't want to brick my water heater ;)

@roboticsalign 
I consulted the installation manual and couldn't find a hint that DIG2 and DIG3 couldn't be used together (I mixed up the DIG-Numbering above). Also, the parameters are different.

I think that's the route I'm going to persue for the moment, as soon as someone has new info we can exchange here

Regards!

**Reply by roboticsalign** [reply]
@Hansfreddy to activate the Dig input, you need to set P16/P23/P24. You can do it, but it won't show the expected behavior. It is not documented, but a confirmed behavior by Daikin support. Sad, because this was an argument for me to buy this device. I'll never buy Daikin again.

**Reply by Hansfreddy** [reply]
@roboticsalign oh I see - thank you for the information.
Okay I guess then I'll try to go and manipulate the touch buttons.

I got to say I'm underwhelmed by the device as well. It's absolutely not what I expected when I read "Smartgrid ready"...

**Reply by lorbetzki** [reply]
Hi, 
> This is very interesting. My first idea was also to attach to the display but I was afraid to brick the device. Now that I know that we can connect in parallel this seems like the way to go. In theory, we should be able to do everything the display can do, just need to figure out the data...

i remove and openend the display, i found a IC from Shanghai Siproin Microelectronics "SSP485". The Display speak Modbus Ascii. We have fixed values/length. You can find more in my "self-conversation" in the disscusion area. In my repo i take some pictures from the inside of display.

https://github.com/lorbetzki/Daikin-EKHHE/discussions/2

**Reply by jcappaert** [reply]
This is very promising, I've also ordered some RS485 hardware to give this a try and potentially help decode the protocols ...

**Reply by Hansfreddy** [reply]
@lorbetzki @jcappaert Wow this is extremely cool!
What can I help you with? I could create more dumps from the communication?

**Comment by tryguntm** [comment]
I read that the Daikin is the same as the Ferroli EGEA TECH with built-in Wi-Fi. Is it possible to adapt something?

**Reply by jcappaert** [reply]
It seems like in the Daikin the Wifi module is not present, you cannot purchase it seperately, and it's not enabled in the firmware.

**Reply by tryguntm** [reply]
Thank you so much for your reply and your work. Is it possible to update your project with the new guidelines? https://developers.esphome.io/blog/2025/05/14/_schema-deprecations/ 

You're a great person.

**Reply by jcappaert** [reply]
Ok updated that quickly - see if it works

**Reply by tryguntm** [reply]
Strangely it doesn't work: INFO ESPHome 2025.9.3
INFO Reading configuration /config/esphome/daikin-altherma-200.yaml...
WARNING Using `number.NUMBER_SCHEMA` is deprecated and will be removed in ESPHome 2025.11.0. Please use `number.number_schema(...)` instead. If you are seeing this, report an issue to the external_component author and ask them to update it…..

**Reply by jcappaert** [reply]
Ok, please create an issue here then: https://github.com/jcappaert/esphome-daikin-ekhhe/issues
You might have to force esphome to update the repository. The old NUMBER_SCHEMA is not used anymore in this update so this warning is likely coming from you still compiling an old version. It compiles fine for me with 2025.10.3.

**Reply by tryguntm** [reply]
I reinstalled ESPHome 2025.10.3 and it worked.. can we improve anything else?
thank you very much
